onnxscript/rewriter/generic_pattern.py

```diff
@@ -297,20 +299,33 @@ def _match_backward(
             )
             return self.none(starting_node, inspect.currentframe().f_lineno)
 
-        for graph_input, pattern_input in zip(graph_node.inputs, pattern_node.inputs):
-            if len(list(graph_input.uses())) != len(list(pattern_input.uses())):
-                self._hint(
-                    "BACKWARD: one input is used outside the pattern",
-                    "-- pattern",
-                    pattern_node,
-                    "-- model",
-                    graph_node,
-                )
-                return self.none(starting_node, inspect.currentframe().f_lineno)
+        has_predecessor_in_pattern = any(
+            pattern_value.producer() is not None for pattern_value in pattern_node.inputs
+        )
+
+        if has_predecessor_in_pattern:
+            for graph_input, pattern_input in zip(graph_node.inputs, pattern_node.inputs):
+                if len(graph_input.uses()) != len(pattern_input.uses()):
+                    self._hint(
+                        "BACKWARD: one input is used outside the pattern",
+                        "-- pattern",
+                        pattern_node,
+                        "-- model",
+                        graph_node,
+                    )
+                    return self.none(starting_node, inspect.currentframe().f_lineno)
 
         for graph_value, pattern_value in zip(graph_node.inputs, pattern_node.inputs):
-            # TODO(rama): Handle constant-pattern
             pattern_pred = pattern_value.producer()
+            if pattern_pred is not None and len(graph_value.uses()) != len(
```

Comment:
I am confused ... isn't this already handled above? Why this check on the number of uses again?

Reply:
The rest of the loop can be executed for the first input if it satisfies the conditions and there is a case where it should not be done.

Reply:
Fill free to find a better solution. The unit test should catch any weird behaviour.